internal/provider/resources/install/ssh/gcp_install.go

```diff
@@ -0,0 +1,193 @@
+}
+
+// Schema implements resource.ResourceWithImportState.
+func (*gcpSshIamWrite) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		// Note that the TF doc generator clobbers _most_ underscores :(
+		MarkdownDescription: `A Google Cloud SSH installation. 
+		
+Installing SSH allows you to manage access to your servers on Google Cloud.`,
+		Attributes: map[string]schema.Attribute{
+			"project_id": schema.StringAttribute{
+				MarkdownDescription: "The Google Cloud project ID.",
+				Required:            true,
+			},
+			"state": schema.StringAttribute{
+				Computed:            true,
+				MarkdownDescription: installresources.StateMarkdownDescription,
+			},
+			"label": schema.StringAttribute{
+				Computed:            true,
+				MarkdownDescription: "The Google Cloud project's alias (if available)",
+			},
+		},
+	}
+}
+
+func (r *gcpSshIamWrite) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	data := internal.Configure(&req, resp)
+	r.installer = &installresources.Install{
+		ProviderData: data,
+		GetItemPath:  r.getItemPath,
+		GetId:        r.getId,
+		GetItemJson:  r.getItemJson,
+		FromJson:     r.fromJson,
+		ToJson:       r.toJson,
+	}
+	if data == nil {
+		return
+	}
+}
+
+func (r *gcpSshIamWrite) getId(data any) *string {
+	model, ok := data.(*gcpSshIamWriteModel)
+	if !ok {
+		return nil
+	}
+
+	str := model.ProjectId.ValueString()
+	return &str
+}
+
+func (r *gcpSshIamWrite) getItemPath(id string) string {
+	return fmt.Sprintf("integrations/%s/config/%s/gcloud:%s", SshKey, installresources.IamWrite, id)
+}
+
+func (r *gcpSshIamWrite) getItemJson(json any) any {
+	inner, ok := json.(*gcpSshIamWriteApi)
+	if !ok {
+		return nil
+	}
+	return inner.Item
+}
+
+func (r *gcpSshIamWrite) fromJson(id string, json any) any {
+	data := gcpSshIamWriteModel{}
+	jsonv, ok := json.(*gcpSshIamWriteJson)
+	if !ok {
+		return nil
+	}
+
+	data.ProjectId = types.StringValue(id)
+	if jsonv.Label != nil {
+		data.Label = types.StringValue(*jsonv.Label)
+	}
+
+	data.State = types.StringValue(jsonv.State)
+
+	return &data
+}
+
+func (r *gcpSshIamWrite) toJson(data any) any {
+	json := gcpSshIamWriteJson{}
+
+	datav, ok := data.(*gcpSshIamWriteModel)
+	if !ok {
+		return nil
+	}
+
+	if !datav.Label.IsNull() && !datav.Label.IsUnknown() {
+		label := datav.Label.ValueString()
+		json.Label = &label
+	}
+
+	// can omit state here as it's filled by the backend
+	return &json
+}
+
+// Create implements resource.ResourceWithImportState.
+func (s *gcpSshIamWrite) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var plan gcpSshIamWriteModel
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	throwaway_response := struct{}{}
+	err := s.installer.ProviderData.Post("integrations/ssh/config", struct{}{}, &throwaway_response)
```

Comment:
can you move this duplicated code to `common.go`?

Reply:
Fixed 👍